numaprom/udsink/train.py

```diff
@@ -141,27 +143,68 @@ def train(datums: List[Datum]) -> Responses:
         # TODO change this to just use **composite_keys
         skeys = [payload.composite_keys["namespace"], payload.composite_keys["name"]]
 
-        version = save_model(
-            skeys=skeys, dkeys=["preproc"], model=preproc_clf, artifact_type="sklearn"
-        )
-        _LOGGER.info(
-            "%s - Preproc model saved with skeys: %s with version: %s", payload.uuid, skeys, version
-        )
-
-        version = save_model(skeys=skeys, dkeys=[model_cfg.name], model=anomaly_model)
-        _LOGGER.info(
-            "%s - Model saved with skeys: %s with version: %s", payload.uuid, skeys, version
-        )
-
-        version = save_model(
-            skeys=skeys, dkeys=["thresh"], model=thresh_clf, artifact_type="sklearn"
-        )
-        _LOGGER.info(
-            "%s - Threshold model saved with skeys: %s with version: %s",
-            payload.uuid,
-            skeys,
-            version,
-        )
+        # Save main model
+        model_registry = RedisRegistry(client=redis_client)
```

Comment:
we need to have multiple models saving method in Registry with transaction and rollbacks for target state, create an issue on numalogic for this?

Reply:
https://github.com/numaproj/numalogic/issues/196
We have it! :)